validation_framework/deployer/coe/kubernetes.py

```diff
@@ -0,0 +1,279 @@
+from fabric import Result
+import json
+import time
+from threading import Event, Thread
+import re
+from pathlib import Path
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class KubernetesCOE(COEBase):
+
+    def __init__(self, device_config: TargetDeviceConfig, **kwargs):
+        self.engine_folder: str = ''
+        self.device = SSHTarget(device_config)
+        self.nuvla_uuid = ''
+        self.namespace = ''
+        self.certificate_manager_pod = ''
+        self.cred_check_thread: CertificateSignCheck = None
+        super().__init__(self.device, logger, **kwargs)
+
+    def start_engine(self,
+                     uuid: NuvlaUUID,
+                     remove_old_installation: bool = True,
+                     extra_envs: dict = None):
+        """
+        Pulls the correct kubernetes image
+        and then start the pods
+        :param uuid:
+        :param remove_old_installation:
+        :param extra_envs:
+        :return:
+        """
+        idparts = uuid.split('/')
+        self.nuvla_uuid = idparts[1]
+        add_repo_cmd = f'sudo helm repo add {cte.NUVLAEDGE_KUBE_LOCAL_REPO_NAME} {cte.NUVLAEDGE_KUBE_REPO}'
+        result: Result = self.device.run_sudo_command(add_repo_cmd)
+        if result.failed:
+            self.logger.error(f'Could not add repo to helm {cte.NUVLAEDGE_KUBE_REPO}: {result.stderr}')
+        version = "''" if self.nuvlaedge_version == '' else self.nuvlaedge_version
+        install_image_cmd = cte.NUVLAEDGE_KUBE_INSTALL_IMAGE.format(
+            uuid=self.nuvla_uuid,
+            repo=cte.NUVLAEDGE_KUBE_LOCAL_REPO_NAME,
+            chart=cte.NUVLAEDGE_KUBE_LOCAL_CHART_NAME,
+            version=version,
+            hostname=self.device.hostname
+        )
+
+        if self.include_peripherals:
+            for peripheral in self.peripherals:
+                cmd_arg = f' --set peripheralManager{peripheral}=true '
+                install_image_cmd = install_image_cmd + cmd_arg
+
+        envs_configuration: dict = self.engine_configuration.model_dump(by_alias=True)
+        if extra_envs:
+            envs_configuration.update(extra_envs)
+
+        self.logger.info(f'Starting NuvlaEdge with UUID: {self.nuvla_uuid} with'
+                         f' configuration: '
+                         f'\n\n {json.dumps(envs_configuration, indent=4)} \n')
+        self.device.run_sudo_command(install_image_cmd, envs=envs_configuration)
+
+        self.namespace = self.__get_current_nuvlaedge_namespace_running()
+
+        self.cred_check_thread = CertificateSignCheck(device_config=self.device.target_config,
+                                                      namespace=self.namespace)
+        self.cred_check_thread.start()
+
+        self.logger.info('Device start command executed')
+
+    def add_peripheral(self):
+        pass
+
+    def remove_peripheral(self):
+        pass
+
+    def stop_engine(self):
+        self.purge_engine()
```

Comment:
Is it possible to divide the purge into stop and remove? Then we can also test the stop and start process of the nuvlaedge if required at any point

Reply:
Done